crates/egui/src/ui.rs

```diff
@@ -1237,6 +1237,17 @@ impl Ui {
             .get_downcast(ClosableTag::NAME)
             .is_some_and(|tag: &ClosableTag| tag.should_close())
     }
+
+    /// Should this [`Ui`] or any of its parents close?
+    pub fn will_close(&self) -> bool {
```

Comment:
`should_close`  and `will_close` are very easy to confuse. They should both document what their differences are, and when to use which.

Maybe their names can be improved too?